homeassistant/components/template/template_entity.py

```diff
@@ -77,6 +80,7 @@
     {
         vol.Optional(CONF_ATTRIBUTES): vol.Schema({cv.string: cv.template}),
         vol.Optional(CONF_AVAILABILITY): cv.template,
+        vol.Optional(CONF_VARIABLES): cv.SCRIPT_VARIABLES_SCHEMA,
```

Comment:
Don't forget to document this since it will be available for all template entities and not only blueprint templates.